File: katas/8 kyu/invert-values/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def invert(lst):
  _list = []
  for x in lst :
    if x > 0:
      _list.append(-x)
    else:
      _list.append(abs(x))
  return _list
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def invert(lst):
    return [-x for x in lst]
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Invert values")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(invert([1,2,3,4,5]),[-1,-2,-3,-4,-5])
        test.assert_equals(invert([1,-2,3,-4,5]), [-1,2,-3,4,-5])
        test.assert_equals(invert([]), [])

File: katas/8 kyu/invert-values/solution.js
// invert-values
/* ------------------------------- MY SOLUTION ------------------------------ */
function invert(array) {
  return array.map(x => -x)
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Invert array values",() => {
  it("Basic Tests", () => {
    assert.deepEqual(invert([1,2,3,4,5]), [-1,-2,-3,-4,-5]);
    assert.deepEqual(invert([1,-2,3,-4,5]), [-1,2,-3,4,-5]);
    assert.deepEqual(invert([]), []);
    assert.deepEqual(invert([0]), [-0]);
  });
});